utils/schemed-kafka-components-utils/src/main/scala/pl/touk/nussknacker/engine/schemedkafka/schemaregistry/universal/ParsedSchemaSupport.scala

```diff
@@ -61,6 +60,15 @@ class AvroSchemaSupport(kafkaConfig: KafkaConfig) extends ParsedSchemaSupport[Av
 
   override def typeDefinition(schema: ParsedSchema): TypingResult = AvroSchemaTypeDefinitionExtractor.typeDefinition(schema.cast().rawSchema())
 
+  def extractParameters(schema: ParsedSchema)(implicit nodeId: NodeId): ValidatedNel[ProcessCompilationError, List[Parameter]] = {
+    AvroSinkValueParameter(schema.cast().rawSchema()).map(_.toParameters)
+  }
+
+  def prepareMessageFormatter(schema: ParsedSchema, schemaRegistryClient: SchemaRegistryClient): Any => Json = {
```

Comment:
Can't we have one final method for both implementations that will just do: `recordFormatterSupport(schemaRegistryClient).formatMessage(sinkValueEncoder(schema, ValidationMode.lax))`
Also I would rename `sinkValueEncoder` to `formValueEncoder` or sth like this.

Reply:
Yeah, we can make it happen :) As mentioned in comment above `prepareMessageFormatter` is now declared final inside `UniversalSchemaSupport`. I've also renamed `sinkValueEncoder` to `formValueEncoder` as you suggested :+1: